src/libstd/collections/hash/table.rs

```diff
@@ -835,18 +838,27 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     elems_left: usize,
 }
 
+unsafe impl<'a, K: Sync, V: Sync> Sync for IterMut<'a, K, V> {}
+unsafe impl<'a, K: Sync, V: Send> Send for IterMut<'a, K, V> {}
```

Comment:
Hm, @huonw do you think this needs `Sync` on `K` for the iter mut case? If you have an `IterMut` you're guaranteed you have the only reference to the map, so each reference handed out to the keys should be unique, right?